### DISCUSSION TITLE
ByteStreamDriverModel Undefined Build Error With Latest Tutorial

### DISCUSSION THREAD

**Discussion by astroesteban** [nasa/fprime#2290] [Q&A] [answered]
Hi y'all,

I was going through the [Hello World: Integration and Testing With F´ Deployments](https://fprime-community.github.io/fprime-tutorial-hello-world/docs/Deployments.html) page just to get acquainted with the latest F Prime updates (the updates to prime-util are phenomenal by the way) and I keep getting this issue when trying to build the `HelloWorldDeployment`:

```cpp
[ 97%] Generating HelloWorldDeploymentTopologyAppAi.xml, Ports_RateGroupsEnumAi.xml
fpp-to-xml
/workspace/MyProject/HelloWorldDeployment/Top/instances.fpp:108.27
  instance comDriver: Drv.ByteStreamDriverModel base id 0x4000 \
                          ^
error: undefined symbol ByteStreamDriverModel
gmake[3]: *** [HelloWorldDeployment/Top/CMakeFiles/HelloWorldDeployment_Top.dir/build.make:113: HelloWorldDeployment/Top/HelloWorldDeploymentTopologyAppAi.xml] Error 1
gmake[2]: *** [CMakeFiles/Makefile2:8300: HelloWorldDeployment/Top/CMakeFiles/HelloWorldDeployment_Top.dir/all] Error 2
gmake[1]: *** [CMakeFiles/Makefile2:8224: HelloWorldDeployment/CMakeFiles/HelloWorldDeployment.dir/rule] Error 2
gmake: *** [Makefile:2418: HelloWorldDeployment] Error 2
[ERROR] CMake erred with return code 2
```

I'm using the latest F Prime version on `devel` (commit f34d596cf41b8fea594a3313b972c027fc28a3de) and running the following commands:

```sh
cd MyProject/HelloWorldDeployment
fprime-util build -j $(nproc)
```

I also opted for the default responses when creating the component and deployment unless otherwise instructed by the tutorial.

Any thoughts as to what could be going wrong?

**Comment by thomas-bc** [comment]
Ah! Yes, there has been changes in the `devel` branch. You will need to checkout the latest released version of F´ instead. Easiest way would be:
```
cd <project-path>/fprime
git fetch --tags
git checkout v3.3.2
```

**Reply by thomas-bc** [reply, marked as answer]
Or if you want to be working with devel, you can update the definition in your `instances.fpp`:

from
```
instance comDriver: Drv.ByteStreamDriverModel base id 0x4000 \
    type "Drv::TcpClient" \
    at "../../Drv/TcpClient/TcpClient.hpp"
```
to
```
instance comDriver: Drv.TcpClient base id 0x4000
```
Or TcpServer/LinuxUartDriver if you're using the TCPServer/UART option.

You can see how the changes were made to our test projects (RPI/Ref) [here](https://github.com/nasa/fprime/pull/2252/files#diff-6d3e707465cf03f064a09bb0b6e1ef923da65135886555ef56de60cb5f856ed1)

**Reply by astroesteban** [reply]
Ah that makes sense! Got it working now. Thanks for the speedy response @thomas-bc! Much appreciated 👍🏼

**Reply by LeStarch** [reply]
@astroesteban how did you encounter this error?  The tools are supposed to guide the user to a most-recent release tag...which should be consistent between the `fprime-util` and the version of F´  checked out.

Did you intentionally bump F´  to `devel` or how were you running off a dedicated tag?

**Reply by LeStarch** [reply]
Never mind.  We've not released since this feature was implemented.

**Reply by astroesteban** [reply]
Yeah and I had just created a new project with `fprime-util` and defaulted to using the `devel` branch option when it asked what version of fprime I wanted to use.
